Makefile.defaults

```diff
@@ -0,0 +1,37 @@
+ifeq (, $(shell which yq))
+$(error "No yq on PATH, consider doing snap install yq")
+endif
+
+YQ_COMMAND     := yq
+VERSIONS_SRC := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))/versions.yaml
+
+define query
+$(or $(shell $(YQ_COMMAND) '.$(1) | select(. != null)' $(VERSIONS_SRC)), \
+	$(error $(1) is not set, and could not be automatically set from $(VERSIONS_SRC)))
+endef
+
+ubuntu_RELEASE ?= focal
+centos_RELEASE ?= stream-8
+
+ubuntu_amd64_IMAGE_URL := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).amd64.url)
+ubuntu_amd64_IMAGE_CHECKSUM := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).amd64.checksum)
+ubuntu_s390x_IMAGE_URL := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).s390x.url)
+ubuntu_s390x_IMAGE_CHECKSUM := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).s390x.checksum)
+centos_amd64_IMAGE_URL := $(call query,cloudimg.centos.$(centos_RELEASE).amd64.url)
+centos_amd64_IMAGE_CHECKSUM := $(call query,cloudimg.centos.$(centos_RELEASE).amd64.checksum)
+
+CAA_SRC := $(or $(CAA_SRC),$(call query,git.cloud-api-adaptor.url))
```

Comment:
So this value doesn't currently do anything (happy for now to remove it now to avoid confusion). 

My plan is that after this PR I will pick up #1148 which will include driving the docker build process via Make so that it can pass through the CAA_SRC value, which is also currently required for the `GO_VERSION`, `PROTOC_VERSION`, and `RUST_VERSION` values you mentioned in your other comment.

So the end goal being that from the dev machine or CI we could run.

```
make docker-build-podvm # example
```

Make would populate the CAA_SRC, GO_VERSION, PROTOC_VERSION & RUST_VERSION values from the local versions.yaml and then call docker e.g.

```
docker build -t podvm_builder -f Dockerfile.podvm_builder --build-arg CAA_SRC=https://github.com/... --build-arg CAA_SRC_REF=0000...  --build-arg  GO_VERSION=...  --build-arg  PROTOC_VERSION=...  --build-arg  RUST_VERSION=...  .
```

Inside the docker image we would clone the remote CAA repo and install the tooling at the right level specified in versions.yaml but then the remote versions.yaml (which should be the same but might not be) is used for the rest of the process. Or we can choose to use the local files which #1148 will introduce in which case the versions.yaml file used will be the same from start to finish.

Reply:
I would say that pulling the source from remote and then continuing the build is a bad idea in general, as the consequences are quite hard to understand as you end up combining outer Makefile and Docker image with inner Makefile that might not fit together.
Imo we should just switch to build from local source completely and remove `CAA_SRC`.

Reply:
But fine for me to discuss this in the next PR. :)